src/Keboola/DbExtractor/Extractor/Extractor.php

```diff
@@ -44,7 +44,7 @@
 
     public function __construct(array $parameters, array $state, Logger $logger)
     {
-        $this->dataDir = $parameters['data_dir'];
+        $this->dataDir = (string) getenv('KBC_DATADIR');
```

Comment:
Tohle řeší php-component, tak by bylo lepší reusnout hodnotu odtamtud - minimálně ti tady chybí fallback na `/data`...

Reply:
Tady je to pravda trochu nesikovne (chybi tomu ten fallback), ale usnout to z `BaseComponent` to nepujde. Extractor uz od ni nededi.

Jestli ti to neni vylozene proti srsti mam v planu to dnes dotahnout v #94 (chci to predelat aby to byla prave trida extraktoru, ktera z BaseComponent bude dedit, takze to bude ok)

Reply:
No a nemůže to dostat ten extractor jako parametr construktoru? Proč musí vědět, kde to je? 

Logika většinou je:

* máš run.php - na ten nesaháš vůbec
* máš Component - tam zkonvertuješ věci co jsou v configu a KBC specific na nějaké obecné
* máš exekutivní třídu, která v izolaci zkonvertuje vstupní parametry na nějaký "splněný úkol" - mělo by být snadné ji vzít a použít třeba unvitř connection na něco - neměla by spoléhat na cokoli co dělá docker runner IMO. 

Tady ti do ni leakuje KBC. 

A extendovat třídu extraktoru z BaseComponent mi nepřijde úplně správně (vzhledem k výše zmíněnému). Ale možná to jen špatně chápu.

Reply:
Parametrem by to bylo fajn, ale to by se musela upravit implementace vsech ostatnich extraktoru. At uz by to slo konstruktorem (i pres to, ze se trida vytvari z promenne) nebo setterem. Proto bych to uz tady neresil, cele to stejne bude jeste jinak.

Na tom podedeni BaseComponenty nevidim nic spatneho, protoze se to prave stara o to ziskani parametru v zavislosti na prostredi. Spise si tu konkretni extraktorovou tridu predstav prave jako ten `Component` co popisujes.

Reply:
Ale celkove bych to spise resil v tom dalsim PR.